test/unit/dialects/abstract/query-generator.test.js

```diff
@@ -98,6 +98,12 @@ describe('QueryGenerator', () => {
       QG.handleSequelizeMethod(this.sequelize.where(this.sequelize.col('foo'), Op.not, null))
         .should.be.equal('foo IS NOT NULL');
     });
+
+    it('should correctly escape $ in sequelize.fn arguments', function() {
+      const QG = getAbstractQueryGenerator(this.sequelize);
+      QG.handleSequelizeMethod(this.sequelize.fn('upper', '$user'))
+        .should.include('$$user');
```

Comment:
And why this test checks for only `$$`? Couldn't it check for `$$$`? Asking out of curiosity, I approve this PR regardless :)